src/sync.cpp

```diff
@@ -640,7 +640,10 @@ Sync::~Sync()
 
     // Create a committer to ensure we update the transfer database in an efficient single commit,
     // if there are transactions in progress.
-    DBTableTransactionCommitter committer(client->tctable);
+    if (client)
+    {
+        DBTableTransactionCommitter committer(client->tctable);
```

Comment:
This commiter becomes out of scope in the next line, so it will not be valid at the point where it's needed (`proctree()` and the delete of `localroot`. 
Since it's very critical to manage the lifetime of be object to properly manage the transaction, you must avoid a `unique_ptr`. There are 2 options, create one with `new` and ensure you call `delete`, or simply declare a committer in each place as required